clippy_lints/src/utils/paths.rs

```diff
@@ -42,6 +42,9 @@ pub const FILE_TYPE: [&str; 3] = ["std", "fs", "FileType"];
 pub const FMT_ARGUMENTS_NEW_V1: [&str; 4] = ["core", "fmt", "Arguments", "new_v1"];
 pub const FMT_ARGUMENTS_NEW_V1_FORMATTED: [&str; 4] = ["core", "fmt", "Arguments", "new_v1_formatted"];
 pub const FMT_ARGUMENTV1_NEW: [&str; 4] = ["core", "fmt", "ArgumentV1", "new"];
+pub const FN: [&str; 3] = ["core", "ops", "Fn"];
+pub const FN_MUT: [&str; 3] = ["core", "ops", "FnMut"];
+pub const FN_ONCE: [&str; 3] = ["core", "ops", "FnOnce"];
```

Comment:
I would prefer to not hardcode those here in this case. The previous implementation managed to avoid those without explicitly checking those paths, was there something wrong about it?

Reply:
This pr is mainly about handling these special cases, isn't it? If outer lifetimes are found in any trait bound involving these three, the lint does not apply. Other traits are fine.

Reply:
After looking at the rewrite, I think it makes sense having those added.